### PR TITLE
OSSM sidecar update

### PR DESCRIPTION
**Pull request by neal-timpe**
For 1.0.1.

### CODE REVIEW

File: modules/ossm-update-app-sidecar.adoc [hunk 1]
```diff
@@ -0,0 +1,16 @@
+// Module included in the following assemblies:
+//
+// * service_mesh/service_mesh_install/installing-ossm.adoc
+
+[id="ossm-update-app-sidecar_{context}"]
+= Updating your application pods
+
+For {ProductName} 1.0.1, run the following commands to update the sidecars on your application pods to the most recent version.
```
Comment (rcernich) [inline modules/ossm-update-app-sidecar.adoc]: I think this should probably say something like,

```md
If updating from a previous version of {ProductName}, you will need to update the sidecar containers on your application pods to the new version.  You can run the following command to trigger a rolling update for deployments configured to use automatic injection:

$ oc patch deployment/<deployment> -p '{"spec":{"template":{"metadata":{"annotations":{"kubectl.kubernetes.io/restartedAt": "'$(date -Iseconds)'"}}}}}'

This simply updates an annotation on the deployment's pod template, which will trigger a new rollout, with updated sidecar containers.
```

Comment (brian-avery) [reply modules/ossm-update-app-sidecar.adoc]: My guess is this only triggers a rolling update with the rolling update strategy (default). There's also the option to use the recreate strategy. https://docs.openshift.com/container-platform/3.3/dev_guide/deployments/deployment_strategies.html#rolling-strategy

Comment (rcernich) [reply modules/ossm-update-app-sidecar.adoc]: good point.  maybe we should just state that this will trigger a redeployment of the existing pods.

Comment (neal-timpe) [reply modules/ossm-update-app-sidecar.adoc]: Ok, I've made the following updates. 

```diff
-For {ProductName} 1.0.1, run the following commands to update the sidecars on your application pods to the most recent version. 
+If you're updating {ProductName} from a previous version, you must update the sidecars on your application pods to the most recent version. 
```

```diff
-Then run following command to trigger a rolling restart of the deployment
+Then, run following command to redeploy the pods:
```

Does that work for you @rcernich and @brian-avery ?

Comment (brian-avery) [reply modules/ossm-update-app-sidecar.adoc]: Sounds good to me. Thanks

**Review by brian-avery — APPROVED** [review]

**Comment by neal-timpe** [comment]
@gbaufake @mattmahoneyrh Can you take a look?

File: modules/ossm-update-app-sidecar.adoc [hunk 2]
```diff
@@ -0,0 +1,16 @@
+// Module included in the following assemblies:
+//
+// * service_mesh/service_mesh_install/installing-ossm.adoc
+
+[id="ossm-update-app-sidecar_{context}"]
+= Updating your application pods
+
+If you're updating {ProductName} from a previous version, you must update the sidecars on your application pods to the most recent version. 
+
+If your deployment uses xref:../service_mesh_day_two/prepare-to-deploy-applications-ossm.adoc#ossm-automatic-sidecar-injection_deploying-applications-ossm[Automatic sidecar injection], you can update the pod template in the deployment by adding or modifying an annotation. Then, run following command to redeploy the pods:
```
[the remark this answers is not in the document]
Comment (neal-timpe) [reply modules/ossm-update-app-sidecar.adoc]: How about: 

>Run the following command to redeploy the pods:

?

[the remark this answers is not in the document]
Comment (rcernich) [reply modules/ossm-update-app-sidecar.adoc]: works for me

**Comment by neal-timpe** [comment]
@tiago @yxun Can you take a look?

File: modules/ossm-update-app-sidecar.adoc (hunk 2 above)
[the remark this answers is not in the document]
Comment (neal-timpe) [reply modules/ossm-update-app-sidecar.adoc]: @rcernich Great. Made that commit. Can you hit the approve review button when you get a chance?

**Review by rcernich — APPROVED** [review]

**Review by yxun — APPROVED** [review]

**Comment by gbaufake** [comment]
@neal-timpe 
LGTM

**Comment by neal-timpe** [comment]
I made a small change that adds a sentence describing the update process. Can you take another look?

File: modules/ossm-update-app-sidecar.adoc [hunk 3]
```diff
@@ -0,0 +1,16 @@
+// Module included in the following assemblies:
+//
+// * service_mesh/service_mesh_install/installing-ossm.adoc
+
+[id="ossm-update-app-sidecar_{context}"]
+= Updating your application pods
+
+If you selected the Automatic Approval Strategy when you were installing your Operators, they update automatically. If you're updating {ProductName} from a previous version, you must update the sidecars on your application pods to the most recent version.
```
Comment (brian-avery) [inline modules/ossm-update-app-sidecar.adoc]: is it better to say "then the operators update automatically"? The sidecars have to be updated regardless.

Comment (neal-timpe) [reply modules/ossm-update-app-sidecar.adoc]: done

Comment (neal-timpe) [reply modules/ossm-update-app-sidecar.adoc]: @brian-avery I rewrote that first sentence with some input from Kevin.

File: modules/ossm-update-app-sidecar.adoc [hunk 4]
```diff
@@ -0,0 +1,16 @@
+// Module included in the following assemblies:
+//
+// * service_mesh/service_mesh_install/installing-ossm.adoc
+
+[id="ossm-update-app-sidecar_{context}"]
+= Updating your application pods
+
+If you selected the Automatic Approval Strategy when you were installing your Operators, then they update automatically. If you're updating {ProductName} from a previous version, you must update the sidecars on your application pods to the most recent version.
```
[the remark this answers is not in the document]
Comment (neal-timpe) [reply modules/ossm-update-app-sidecar.adoc]: @tvieira Good idea. I just added that link.

**Review by rcernich — APPROVED** [review]

**Review by tvieira — APPROVED** [review]
LGTM

**Comment by neal-timpe** [comment]
@openshift/team-documentation Can you review this PR for a release today?

File: modules/ossm-update-app-sidecar.adoc [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+
+If you selected the xref:../../operators/olm-adding-operators-to-cluster.adoc#olm-installing-operators-from-operatorhub_olm-adding-operators-to-a-cluster[Automatic Approval Strategy] when you were installing your Operators, then the Operators update the control plane automatically, but not your applications. Existing applications continue to be part of the mesh and function accordingly. The application administrator must restart applications to upgrade the sidecar.
+
+If your deployment uses xref:../service_mesh_day_two/prepare-to-deploy-applications-ossm.adoc#ossm-automatic-sidecar-injection_deploying-applications-ossm[Automatic sidecar injection], you can update the pod template in the deployment by adding or modifying an annotation. Run the following command to redeploy the pods:
+
+----
+$ oc patch deployment/<deployment> -p '{"spec":{"template":{"metadata":{"annotations":{"kubectl.kubernetes.io/restartedAt": "'`date -Iseconds`'"}}}}}'
+----
+
+If your deployment doesn't use automatic sidecar injection, you must manually update the sidecars by modifying the sidecar container image specified in the deployment or pod.
```
Comment (mburke5678) [inline modules/ossm-update-app-sidecar.adoc]: s/doesn't/does not

Comment (neal-timpe) [reply modules/ossm-update-app-sidecar.adoc]: Done and squashed @mburke5678

**Comment by kalexand-rh** [comment]
@neal-timpe says this is ready for production. Merging.

**Pull request merged**